.github/workflows/CI.yml

```diff
@@ -85,14 +85,14 @@ jobs:
     runs-on: ubuntu-latest
     needs: lint
     container:
-      image: nicolasbock/bml:master
+      image: nicolasbock/bml:pr-454
```

Comment:
Is it correct?

Reply:
Yes. Let me explain:

The GitHub workflow has a step in it that builds the CI image and pushes that to Docker Hub. All other jobs in the CI workflow pull these images from Docker Hub and run their tests in them.

Ideally we would use the image built in a PR for the tests in that same PR so we can test the image and run our tests using that image. Unfortunately this is not possible with GitHub Actions. What we do instead is to tag the image with the branch so that PRs and the master branch are protected from each other. But that means that in order to use the image built in a PR we need to manually change the tag in the workflow file as I have done here.

Once we merge this PR I need to submit a follow up PR to change the tag back to `master`.

Reply:
I trust you. I was just making sure it was correct.

Reply:
Thanks for pointing out that code change! That's the kind of change we need to watch out for because it can alter the CI process.

Maybe we should think about not using docker images so we can alter the test environment within a PR more easily. I'll have a look at that.